pennylane/transforms/batch_transform.py

```diff
@@ -158,9 +159,60 @@ def __init__(self, transform_fn, expand_fn=None, differentiable=True):
         self.transform_fn = transform_fn
         self.expand_fn = expand_fn
         self.differentiable = differentiable
+        self.qnode_wrapper = self.default_qnode_wrapper
         functools.update_wrapper(self, transform_fn)
 
-    def qnode_execution_wrapper(self, qnode, targs, tkwargs):
+    def custom_qnode_wrapper(self, fn):
+        """Register a custom QNode execution wrapper function
+        for the batch transform.
+
+        **Example**
+
+        .. code-block:: python
+
+            def my_transform(tape, *targs, **tkwargs):
+                ...
+                return tapes, processing_fn
+
+            @my_transform.custom_qnode_wrapper
+            def my_custom_qnode_wrapper(self, qnode, targs, tkwargs):
+                def wrapper_fn(*args, **kwargs):
+                    # construct QNode
+                    qnode.construct(args, kwargs)
+                    # apply transform to QNode's tapes
+                    tapes, processing_fn = self.construct(qnode.qtape, *targs, **tkwargs)
+                    # execute tapes and return processed result
+                    ...
+                    return processing_fn(results)
+                return wrapper_fn
+
+        The custom QNode execution wrapper must have arguments
+        ``self`` (the batch transform object), ``qnode`` (the input QNode
+        to transform and execute), ``targs`` and ``tkwargs`` (the transform
+        arguments and keyword arguments respectively).
+
+        It should return a callable object that accepts the *same* arguments
+        as the QNode, and returns the transformed numerical result.
+
+        The default :meth:`~.default_qnode_wrapper` method may be called
+        if only pre- or post-processing dependent on QNode arguments is required:
+
+        .. code-block:: python
+
+            @my_transform.custom_qnode_wrapper
+            def my_custom_qnode_wrapper(self, qnode, targs, tkwargs):
+                transformed_qnode = self.default_qnode_wrapper(qnode)
+
+                def wrapper_fn(*args, **kwargs):
+                    args, kwargs = pre_process(args, kwargs)
+                    res = transformed_qnode(*args, **kwargs)
+                    ...
+                    return ...
+                return wrapper_fn
+        """
+        self.qnode_wrapper = types.MethodType(fn, self)

```

Comment:
Never seen `types` before, so this is what performs the "replacement" with the custom wrapper?

Reply:
I hadn't either, I found this module from a stackoverflow answer while writing this PR 😆 

Essentially, what `types.MethodType` does is converts a standard Python function into a method of a particular object. This is required, since methods act slightly differently to functions (Python 'injects' `self` as the first argument in methods, but not in functions).

So it is a safer way of overriding an instance method at runtime, as opposed to naively doing

```python
>>> a = SomeClass()
>>> a.method = my_func  # won't work in all edge cases
>>> a.method = MethodType(my_func, a)  # safer
```